addon/session-stores/cookie.js

```diff
@@ -172,23 +191,30 @@ export default BaseStore.extend({
   },
 
   _calculateExpirationTime() {
-    let cachedExpirationTime = this._read(`${this.cookieName}-expiration_time`);
+    let cachedExpirationTime = this._read(`${this.get('cookieName')}-expiration_time`);
     cachedExpirationTime     = !!cachedExpirationTime ? new Date().getTime() + cachedExpirationTime * 1000 : null;
-    return !!this.cookieExpirationTime ? new Date().getTime() + this.cookieExpirationTime * 1000 : cachedExpirationTime;
+    return this.get('cookieExpirationTime') ? new Date().getTime() + this.get('cookieExpirationTime') * 1000 : cachedExpirationTime;
   },
 
   _write(value, expiration) {
     let cookieOptions = {
-      domain:  this.cookieDomain,
+      domain:  this.get('cookieDomain'),
       expires: isEmpty(expiration) ? null : new Date(expiration),
       path:    '/',
       secure:  this.get('_secureCookies')
     };
-    this.get('_cookies').write(this.cookieName, value, cookieOptions);
+    if (this._oldCookieName) {
+      A([this._oldCookieName, `${this._oldCookieName}-expiration_time`]).forEach((oldCookie) => {
+        let expires     = new Date(0);
+        this.get('_cookies').write(oldCookie, null, assign({}, cookieOptions, { expires }));
```

Comment:
This could actually use ember-cookies' [`clear` method](https://github.com/simplabs/ember-cookies#api) to delete the cookie.